webpreview/previews.py

```diff
@@ -37,32 +37,39 @@ def __init__(self, url = None, properties = None, timeout=None, headers=None):
         except MissingSchema: # if no schema add http as default
             url = "http://" + url
 
-        # throw URLUnreachable exception for just incase
-        try:
-            res = requests.get(url, timeout=timeout, headers=headers)
-        except (ConnectionError, HTTPError, Timeout, TooManyRedirects):
-            raise URLUnreachable("The URL is unreachable.")
-
-        if res.status_code == 404:
-            raise URLNotFound("The web page does not exist.")
+        # if content is provided don't fetch from url
+        if not content:
+            content = PreviewBase.get_content(url)
 
         # its safe to assign the url
         self.url = url
 
-
         if not properties:
             raise EmptyProperties("Please pass list of properties to be extracted.")
         # its safe to assign properties
         self.properties = properties
-        self._soup = BeautifulSoup(res.text, "html.parser")
+        self._soup = BeautifulSoup(content, "html.parser")
+
+    @staticmethod
+    def get_content(url, timeout, headers):
+        # throw URLUnreachable exception for just incase
+        try:
+            res = requests.get(url, timeout=timeout, headers=headers)
+        except (ConnectionError, HTTPError, Timeout, TooManyRedirects):
+            raise URLUnreachable("The URL is unreachable.")
+
+        if res.status_code == 404:
+            raise URLNotFound("The web page does not exist.")
 
+        return res.text
 
 class GenericPreview(PreviewBase):
     """
     Extracts title, description, image from a webpage's body instead of the meta tags.
     """
-    def __init__(self, url = None, properties = ['title', 'description', 'image'], timeout=None, headers=None):
-        super(GenericPreview, self).__init__(url, properties, timeout=timeout, headers=headers)
+    def __init__(self, url = None, properties = ['title', 'description', 'image'], timeout=None, headers=None,
+                 content=None):
```

Comment:
Can you remove the line break there?

Reply:
Thank you @ludbek for review comments. I have removed the newline and added content param in README.md. Kindly review.